### DISCUSSION TITLE
Interval timer at sunrise/sunset

### DISCUSSION THREAD

**Discussion by candanga** [arendst/Tasmota#16072] [Q&A] [answered]
Hi, I’m trying to make a Sonoff Basic with Tasmota, switch ON/OFF after X amount of minutes.

i.e. At Sunrise, POWER ON 20 minutes and POWER OFF 30 minutes (repeat), and at Sunset, Power ON 15 minutes and POWER OFF for 2 hours (repeat).

I’ve looking at the commands, rules and timers; trying to adapt the following project with no luck:
https://docodethatmatters.com/automate-sprinkler-with-tasmota/

These one-shot timers are set to start only for an ammount of time and that's it.

Timer1 {"Enable":1,"Mode":1,"Time":"00:00","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":1}
Timer2 {"Enable":1,"Mode":1,"Time":"00:30","Window":0,"Days":"1111111","Repeat":1,"Output":1,"Action":0}

Don't know how to merge them with poweronstate 5 or pulsetime

Any help in the right direction will be greatly appreciated! : )

**Comment by sfromis** [comment, marked as answer]
If you let the timers have action 3 rule, you can create rules for more detailed control than just on/off.

**Reply by candanga** [reply]
Thanks for the suggestion, but I’m a complete noob on how to apply this. I can see in the documentation that Rule 3 is “RULE/BLINK”. How do I determine the time and make the rule know when its sunrise/sunset to apply different intervals? 

Thanks!
